src/main/java/com/tozny/sdk/realm/methods/user_push/UserPushRequest.java

```diff
@@ -0,0 +1,54 @@
+package com.tozny.sdk.realm.methods.user_push;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.tozny.sdk.ToznyApiRequest;
+
+import javax.annotation.Nullable;
+
+/**
+ * Collects request parameters for invoking the "realm.user_push" method.
+ * Requires that either the `user_id` or `email` or `username` field be set.
+ */
+@JsonAutoDetect(getterVisibility=JsonAutoDetect.Visibility.NONE)
+public class UserPushRequest implements ToznyApiRequest {
+
+    @JsonProperty private String method = "realm.user_push";
+    @JsonProperty private String session_id;
+    @JsonProperty private String user_id;
+    @JsonProperty private String email;
+    @JsonProperty private String username;
+
+    public UserPushRequest(String session_id, @Nullable String user_id, @Nullable String email, @Nullable String username) {
+        if (user_id == null && email == null && username == null) {
```

Comment:
If they're all blank or whitespace, the API itself will return an exception that is caught by the SDK and converted to a `ToznyApiException`.